io/include/traccc/io/read_geometry.hpp

```diff
@@ -27,4 +27,19 @@ namespace traccc::io {
 geometry read_geometry(std::string_view filename,
                        data_format format = data_format::csv);
 
+/// Read in the detector geometry description from a detector object
+template <typename detector_t>
+geometry alt_read_geometry(const detector_t& det) {
```

Comment:
So , this is not in fact using the detray geometry, but just copying the data frm detray into a traccc data structure?

Reply:
That's what this function is doing for upstream algorithms (clusterization and seeding).
But track fitting or finding uses `detray::detector`

Reply:
I think, if possible, we should be able to look the transforms up directly in the detector, instead of copying them and  then searching for them